packages/store/src/Schema.sol

```diff
@@ -169,39 +169,32 @@ library SchemaInstance {
     if (_numTotalFields > MAX_TOTAL_FIELDS) revert SchemaLib.SchemaLib_InvalidLength(_numTotalFields);
 
     // No static field can be after a dynamic field
-    uint256 countStaticFields;
-    uint256 countDynamicFields;
     uint256 _staticDataLength;
-    for (uint256 i; i < _numTotalFields; ) {
+    for (uint256 i; i < _numStaticFields; ) {
+      uint256 staticByteLength = schema.atIndex(i).getStaticByteLength();
+      if (staticByteLength == 0) {
+        revert SchemaLib.SchemaLib_StaticTypeAfterDynamicType();
+      }
+      _staticDataLength += staticByteLength;
+      unchecked {
+        i++;
+      }
+    }
+
+    for (uint256 i = _numStaticFields; i < _numTotalFields; ) {
       uint256 staticByteLength = schema.atIndex(i).getStaticByteLength();
       if (staticByteLength > 0) {
-        _staticDataLength += staticByteLength;
-        // Static field in dynamic part
-        if (i >= _numStaticFields) revert SchemaLib.SchemaLib_StaticTypeAfterDynamicType();
-        unchecked {
-          countStaticFields++;
-        }
-      } else {
-        // Dynamic field in static part
-        if (i < _numStaticFields) revert SchemaLib.SchemaLib_StaticTypeAfterDynamicType();
-        unchecked {
-          countDynamicFields++;
-        }
+        revert SchemaLib.SchemaLib_StaticTypeAfterDynamicType();
       }
       unchecked {
         i++;
       }
     }
+
     // Static length sums must match
     if (_staticDataLength != schema.staticDataLength()) {
       revert SchemaLib.SchemaLib_InvalidLength(schema.staticDataLength());
     }
```

Comment:
should we move this check up so we can revert before the second loop starts?

Reply:
Yes, that's clearer IMO because it's the check is about static data. And wastes less computation if it reverts